src/core_plugins/kbn_vislib_vis_types/public/controls/point_series/series.js

```diff
@@ -0,0 +1,80 @@
+import _ from 'lodash';
+import uiModules from 'ui/modules';
+import vislibSeriesTemplate from 'plugins/kbn_vislib_vis_types/controls/point_series/series.html';
+const module = uiModules.get('kibana');
+
+module.directive('vislibSeries', function ($parse, $compile) {
+  return {
+    restrict: 'E',
+    template: vislibSeriesTemplate,
+    replace: true,
+    link: function ($scope) {
+      function makeSerie(label) {
+        const last = $scope.series[$scope.series.length - 1];
+        return {
+          show: true,
+          mode: last ? last.mode : 'normal',
+          type: last ? last.type : 'line',
+          drawLinesBetweenPoints: true,
+          showCircles: true,
+          interpolate: 'linear',
+          data: {
+            label: label
+          },
+          valueAxis: $scope.vis.params.valueAxes[0].id
+        };
+      }
+
+      $scope.series = $scope.vis.params.seriesParams;
+      $scope.$watch(() => {
+        return $scope.vis.aggs.map(agg => {
+          try {
+            return agg.makeLabel();
+          } catch (e) {
+            return '';
+          }
+        }).join();
+      }, () => {
+        let serieCount = 0;
+        const schemaTitle = $scope.vis.type.schemas.metrics[0].title;
+        $scope.vis.aggs.forEach(agg => {
+          if (!agg.type) return;
+          if (agg.schema.title !== schemaTitle) return;
+          if (agg.type.type !== 'metrics') return;
+          if ($scope.vis.params.seriesParams[serieCount]) {
+            $scope.vis.params.seriesParams[serieCount].data.label = agg.makeLabel();
+          } else {
+            const serie = makeSerie(agg.makeLabel());
+            $scope.vis.params.seriesParams.push(serie);
+          }
+          serieCount++;
+        });
+        $scope.vis.params.seriesParams = $scope.vis.params.seriesParams.slice(0, serieCount);
+      });
+
+      $scope.$watch(() => {
+        return $scope.vis.params.seriesParams.map(series => series.type).join();
+      }, () => {
+        const types = _.uniq(_.map($scope.vis.params.seriesParams, 'type'));
+        $scope.savedVis.type = types.length === 1 ? types[0] : 'histogram';
```

Comment:
when we are saving the vis we want to update the type (so if you started with line chart and you then changed it to bar chart we want bar chart icon to appear)

i am not sure this condition is necessary but if you would to save a chart with nothing on it ... it would save as histogram.